iso15118/secc/states/iso15118_2_states.py

```diff
@@ -408,6 +409,28 @@ async def process_message(
 
         service_detail_req: ServiceDetailReq = msg.body.service_detail_req
 
+        is_found = False
+        for service_details in self.comm_session.offered_services:
+            if service_detail_req.service_id == service_details.service_id:
+                is_found = True
+                break
+
+        if not is_found:
+            # [V2G2-425] The SECC shall respond with a negative ResponseCode
+            # 'FAILED_ServiceIDInvalid'
+            # if the EVCC provided a not previously retrieved ServiceID in the
+            # ServiceDetailsReq.
+            msg_type = get_msg_type(str(message))
```

Comment:
As discussed, this is perhaps simpler. Also returns the service id of the failed ServiceDetailReq

```
            error_service_detail_res = ServiceDetailRes(
                response_code=ResponseCode.FAILED_SERVICE_ID_INVALID,
                service_id=service_detail_req.service_id,
            )
            self.create_next_message(
                None,
                error_service_detail_res,
                Timeouts.V2G_SECC_SEQUENCE_TIMEOUT,
                Namespace.ISO_V2_MSG_DEF,
            )
            return
```

Reply:
Done